pkg/cloudprovider/cloudapi/azure/azure_security.go

```diff
@@ -320,53 +337,39 @@ func (computeCfg *computeServiceConfig) buildEffectiveNSGSecurityRulesToApply(ap
 			continue
 		}
 
-		// Nephe rules will be created from ruleStartPriority and have description.
+		// check if the rule is Nephe rules based on ruleStartPriority and description.
 		desc, ok := securitygroup.ExtractCloudDescription(rule.Properties.Description)
 		if *rule.Properties.Priority >= ruleStartPriority && ok {
-			// skip any rules created by current processing appliedToGroup (as we have new rules for this group).
+			// check if the rule is created by current processing appliedToGroup.
 			ruleAddrGroupName := desc.AppliedToGroup
 			_, _, isNepheControllerCreatedRule := securitygroup.IsNepheControllerCreatedSG(desc.AppliedToGroup)
 			if isNepheControllerCreatedRule && strings.Compare(ruleAddrGroupName, appliedToGroupNepheControllerName) == 0 {
-				continue
+				// if the rule no longer exists in newRules then skip its update.
+				// if the rule exists then remove it from newRules and add to current rules.
+				newRules := newIngressSecurityRules
+				if *rule.Properties.Direction == armnetwork.SecurityRuleDirectionOutbound {
+					newRules = newEgressSecurityRules
+				}
+				idx, found := findSecurityRule(rule, newRules)
```

Comment:
Here you are checking if the newRule and currentRule realized in cloud are same. 
If they are same, then you are going to skip updating the rule right?

Reply:
Yes. However this logic changes in the new revision